fileutil/file.go

```diff
@@ -41,6 +41,8 @@ func CouldBeScript(info os.FileInfo) ScriptConfidence {
 	switch {
 	case info.IsDir(), name[0] == '.':
 		return ConfNotScript
+	case info.IsDir(), info.Mode()&os.ModeSymlink == os.ModeSymlink:
+		return ConfNotScript
```

Comment:
Note that the commas here are OR, not AND. So `info.IsDir()` is a no-op since it's caught above.

Effectively this discards all symlinks, so it should be wrong. If you add tests for both symlinks to files and symlinks to dirs, this change should break symlinks to files - which I believe we want to keep working.

Also, you can simplify the second part to `info.Mode()&os.ModeSymlink != 0`.

Reply:
Re: OR/AND: Thanks, that was a bad assumption on my part.

Re: discarding symlinks: Doesn't testing symlinks result in testing the same file multiple times? Also, doesn't that possibly end up testing (and possibly modifying in-place) files outside the context provided by the user?

Reply:
Yes, that can potentially lead to doing the same files multiple times. Which is OK for now since we don't parse multiple files at once, but it could be problematic sometime in the future if we're not careful.

Regarding files outside the context, I guess that depends on how you use symlinks and how you expect tools to treat them. I'd say either using them or discarding them would be OK things to do.

But in general I agree - trying to follow symlinks can lead to tricky situations and unexpected behavior for little benefit. Simply skipping all symlinks seems like a good approach for now. I'd rather be conservative and only reconsider the logic if anyone raises an issue about it.

Tests are less important if we just skip all symlinks, but they'd still be useful IMO. As said, feel free to leave that for me.

Reply:
I pushed another commit intending to drop all symlinks. I left `info.Mode()&os.ModeSymlink == os.ModeSymlink` instead of replacing it with `info.Mode()&os.ModeSymlink != 0` because I find the former more readable.